src/delay.ml

```diff
@@ -307,6 +307,74 @@ let parse_durationl l =
   let s = String.concat " " l in
   parse_duration s
 
+let duration_of_daypack_duration (x : Daypack_lib.Duration.t) : duration =
+  C.Period.lmake ~day:x.days ~hour:x.hours ~minute:x.minutes ~second:x.seconds ()
+
+let get_current_tz_offset_s () =
+  Ptime_clock.current_tz_offset_s () |> Option.get |> Option.some
+
+let search_param =
+  let open Daypack_lib in
+  Search_param.Years_ahead_start_unix_second {
+    search_using_tz_offset_s = get_current_tz_offset_s ();
+    start = Time.Current.cur_unix_second ();
+    search_years_ahead = 5;
+  }
+
+let parse_durationl' l =
+  let s = String.concat " " l in
+  Daypack_lib.Duration.of_string s
+  |> Result.map duration_of_daypack_duration
```

Comment:
I did, but now test fails. :(

Reply:
Yeah I definitely should have removed the dead code, my apologies.

I'm checking the test results now.

Reply:
The test results seem to make more sense after feeding `d` to `Delay.Iso8601.parse` instead of `s`

```
  [OK]          Periods          0   PT1S.
  [OK]          Periods          1   PT1H15M.
> [FAIL]        Periods          2   P1YT2M.
  [FAIL]        Periods          3   P2M3DT23M.
  [FAIL]        Periods          4   PT2M1S.
  [FAIL]        Periods          5   P1Y3DT2H.
  [OK]          Periods          6   PT3M30s.
  [OK]          Periods          7   PT3M30s.
```

The 4 cases of failure come from "year" and "month" not defined in daypack duration expr.

Reply:
Right, year and months ... I must admit, I have not used those.

Ideally I would like to implement a way where I could enter dates into my calendar by calling fugit, and then months/years become useful. But I don't think I've ever used them otherwise.

Reply:
> Ideally I would like to implement a way where I could enter dates into my calendar by calling fugit, and then months/years become useful. But I don't think I've ever used them otherwise.

Huh, okay yeah, I can see why it can be useful...

My main issue was there didn't seem to be a good way to normalise years or months into days/hours/seconds.

But it makes sense in terms of just "bumping" the field by some number, as only an offset for date. Hm....I have not put any thought into that yet, will have to think about that more...

Reply:
What do you reckon if that is left to, say, a `Date_offset` module?

The relationship would be:
- Reduction from `Duration.t` to `Date_offset.t` is well defined
- Reduction from `Date_offset.t` to `Duration.t` is not (always) well defined
- Addition/subtraction of date with `Date_offset.t` is well defined
- Addition/subtraction of date with duration is also well defined (by first point)

Reply:
Is that just for year/month ? I think we can forget that for now, and revisit the idea later.